### ISSUE TITLE
Deserializing JavaScriptObject that contains null property value throws dangling name exception

### ISSUE THREAD

**Issue by richturner** [nmorel/gwt-jackson#106]
I have a JSON payload that represents an Asset object with an attributes property that is of type JsonObject (GWT elemental) and I've created a custom deserializer for this as follows:

```
public class ElementalJsonDeserializer extends JsonDeserializer<JsonObject> {

    @Override
    protected JsonObject doDeserialize(com.github.nmorel.gwtjackson.client.stream.JsonReader reader, JsonDeserializationContext ctx, JsonDeserializerParameters params) {
        if (reader.hasNext()) {
            return reader.nextJavaScriptObject(ctx.isUseSafeEval()).cast();
        }
        return null;
    }
}
```

The outer Asset class is just using the default generated BeanDeserializer.

Problem is if this inner JsonObject contains a property with a null value then an exception is thrown stating "dangling name"

Example JSON payload that causes the exception (throws "dangling name 'value'" exception):

```
{
	"id": "uOHBvCapQ7iy2M1HEZstWQ",
	"createdOn": 1488991970595,
	"name": "Scene",
	"type": "urn:asset:thing",
	"attributes": {
		"Return Schedule": {
			"type": "String[]",
			"value": true,
			"metadata": [{
				"name": "urn:asset:meta:trigger:type",
				"value": "ONCE"
			},
			{
				"name": "urn:asset:meta:trigger:times",
				"value": null
			},  
			{
				"name": "urn:asset:meta:trigger:sceneValues",
				"value": {
					"0": null,
					"1": true,
					"2": true,
					"3": true,
					"4": true
				}
			}]
		},
		"urn:asset:meta:scene:action": {
			"type": "Object[]",
			"value": [{
				"value": "Away Schedule",
				"attributeRef": {
					"attributeName": "ExecuteTrigger"
				}
			},
			{
				"value": false,
				"attributeRef": {
					"attributeName": "Day Schedule"
				}
			},
			{
				"value": false,
				"attributeRef": {
					"attributeName": "Away Schedule"
				}
			},
			{
				"value": false,
				"attributeRef": {
					"attributeName": "Evening Schedule"
				}
			},
			{
				"value": false,
				"attributeRef": {
					"attributeName": "Night Schedule"
				}
			}]
		}
	},
	"path": ["uOHBvCapQ7iy2M1HEZstWQ"]
}
```

I traced the issue down to NonBufferedJsonReader.nextJavaScriptObject which ends up calling nextValue() to get the JsonObject string but problem I see is that nextValue while loop cannot handle PEEKED_NULL (there is no if statement for it) so when a null is encountered the writer is out of sync and writer.close() throws the exception. I cloned this class and added the following statement in the while loop

```
 else if (p == PEEKED_NULL) {
   writer.nullValue();
}
```

And with a bit of hackery I managed to get my custom deserializer to use my hacked JsonReader and with the above code alteration I no longer get the dangling name exception and all works well

**Comment by richturner** [comment]
I was trying to find a nice way of using my own JsonReader to work around this issue but I cannot figure out a way of doing this (comments at the top of NonBufferedJsonReader suggest a custom JsonReader is possible).

**Comment by nmorel** [comment]
It was probably possible on a past release but now, it seems you can't change it.  
The only way would be to override the context but its only constructor is private.

Can you make a PR to add your condition ? And add a test please :)
You juste have to update AbstractJsonReaderTest.testNextJavaScriptObjectNoRootObject for example.

**Comment by nmorel** [comment]
I released a version 0.14.2 with your correction.
Thanks :)

**Comment by richturner** [comment]
Brilliant! Thanks for the quick turn around
